	This trivial patch causes gcc to emit a warning whenever
	it generates a trampoline.  These are otherwise hard to
	locate.  It is rigged to default ON - to have it default
	to OFF remove the text 'Init(1)' from the common.opt
	patch, leaving just 'Common Var(warn_trampolines)'.
	Kevin F. Quinn <kevquinn@gentoo.org> 17 Jan 2006

---
 gcc/builtins.c |    3 +++
 gcc/common.opt |    4 ++++
 2 files changed, 7 insertions(+)

Index: gcc-4.4.1/gcc/common.opt
===================================================================
--- gcc-4.4.1.orig/gcc/common.opt	2009-11-19 13:43:48.000000000 +0100
+++ gcc-4.4.1/gcc/common.opt	2009-11-19 13:45:36.000000000 +0100
@@ -197,6 +197,10 @@
 Common Var(warn_type_limits) Init(-1) Warning
 Warn if a comparison is always true or always false due to the limited range of the data type
 
+Wtrampolines
+Common Var(warn_trampolines) Init(1)
+Warn whenever a trampoline is generated
+
 Wuninitialized
 Common Var(warn_uninitialized) Warning
 Warn about uninitialized automatic variables
Index: gcc-4.4.1/gcc/builtins.c
===================================================================
--- gcc-4.4.1.orig/gcc/builtins.c	2009-11-19 13:43:48.000000000 +0100
+++ gcc-4.4.1/gcc/builtins.c	2009-11-19 13:45:36.000000000 +0100
@@ -5768,6 +5768,9 @@
   trampolines_created = 1;
   INITIALIZE_TRAMPOLINE (r_tramp, r_func, r_chain);
 
+  if (warn_trampolines)
+    warning (OPT_Wtrampolines, "generating trampoline in object (requires executable stack)");
+
   return const0_rtx;
 }
 
